components/AutoComplete/AutoComplete.jsx

```diff
@@ -0,0 +1,409 @@
+import React, { useState, useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import { shadow, normalizeChars } from '../shared';
+import { baseFontSize, colors, spacing } from '../shared/theme';
+import Icon from '../Icon';
+
+import {
+  TextInput,
+  InputLabel,
+  InputErrorMessage,
+  HelperText,
+  RequiredMark,
+} from '../Input/sub-components';
+import useKeyPress from './SubComponents/UseKeyPress';
+
+const ITEM_HEIGHT = '44px';
+const MAX_ITEMS_VISIBILITY = 7;
+const DROPITEM_FONT_SIZE = baseFontSize * 0.875;
+
+const ComponentWrapper = styled.div`
+  ${({
+    theme: {
+      colors: { neutral },
+    },
+    skin = 'default',
+  }) => css`
+    position: relative;
+    color: ${skin === 'default' ? neutral[700] : neutral[0]};
+  `}
+`;
+
+const InputWrapper = styled.div`
+  position: relative;
+`;
+
+const InputText = styled(TextInput)`
+  ${({
+    theme: {
+      spacing: { xsmall },
+    },
+  }) => css`
+    margin-top: ${xsmall}px;
+  `}
+`;
+
+const InputIcon = styled(Icon)`
+  cursor: pointer;
+  position: absolute;
+  ${({
+    theme: {
+      spacing: { xsmall, medium },
+    },
+  }) => css`
+    right: ${medium}px;
+    bottom: ${xsmall * 1.25}px;
+    width: ${baseFontSize * 1.5}px;
+  `}
+`;
+
+const InputErrorIcon = styled(InputIcon).attrs({ name: 'error' })`
+  ${({
+    theme: {
+      colors: {
+        error: { 700: error700 },
+      },
+    },
+    skin,
+  }) => css`
+    color: ${skin === 'default' ? error700 : 'inherit'};
+  `}
+`;
+
+const List = styled.ul`
+  border-radius: 4px;
+  box-sizing: border-box;
+  list-style: none;
+  max-height: calc(${ITEM_HEIGHT} * ${MAX_ITEMS_VISIBILITY});
+  overflow: auto;
+  padding: 0;
+  position: absolute;
+  width: 100%;
+  z-index: 1;
+
+  ${({ theme }) => {
+    const {
+      spacing: { xxsmall },
+      colors: {
+        neutral: { 0: neutral0, 700: neutral700 },
+      },
+    } = theme;
+    return css`
+      background-color: ${neutral0};
+      margin-top: ${xxsmall}px;
+      ${shadow(3, neutral700)({ theme })};
+    `;
+  }}
+`;
+
+const ListIten = styled.li`
```

Comment:
```suggestion
const ListItem = styled.li`
```

<img src="https://media2.giphy.com/media/3o7TKpHfFyh6DWYOKQ/giphy.gif"/>

Reply:
<img src="https://media4.giphy.com/media/Ty9Sg8oHghPWg/giphy.gif"/>